SignalServiceKit/src/Loki/ProofOfWork.swift

```diff
@@ -55,36 +56,19 @@ public enum ProofOfWork {
         }
     }()
     
-    public struct Configuration {
-        var pubKey: String
-        var data: String
-        var timestamp: Date
-        var ttl: Int
-
-        var payload: [UInt8] {
-            let timestampString = String(Int(timestamp.timeIntervalSince1970))
-            let ttlString = String(ttl)
-            let payloadString = timestampString + ttlString + pubKey + data
-            return payloadString.bytes
-        }
-
-        public init(pubKey: String, data: String, timestamp: Date, ttl: Int) {
-            self.pubKey = pubKey
-            self.data = data
-            self.timestamp = timestamp
-            self.ttl = ttl
-        }
-    }
-    
     /// Calculate a proof of work with the given configuration
     ///
     /// Ref: https://bitmessage.org/wiki/Proof_of_work
     ///
-    /// - Parameter config: The configuration
+    /// - Parameters:
+    ///   - data: The message data
+    ///   - pubKey: The message recipient
+    ///   - timestamp: The timestamp
+    ///   - ttl: The message time to live
     /// - Returns: A nonce string or nil if it failed
-    public static func calculate(with config: Configuration) -> String? {
-        let payload = config.payload
-        let target = calcTarget(ttl: config.ttl, payloadLength: payload.count, nonceTrials: nonceTrialCount)
+    @objc public class func calculate(forData data: String, pubKey: String, timestamp: UInt64, ttl: Int) -> String? {
```

Comment:
`static` should be fine too though I've noticed that it won't show up in auto-complete that way